cherab/openadas/rates/radiated_power.pyx

```diff
@@ -46,7 +46,7 @@ cdef class LineRadiationPower(CoreLineRadiationPower):
         extrapolation_type = 'nearest' if extrapolate else 'none'
         self._rate = Interpolator2DArray(np.log10(ne), np.log10(te), rate, 'cubic', extrapolation_type, INFINITY, INFINITY)
 
-    cdef double evaluate(self, double electron_density, double electron_temperature) except? -1e999:
+    cpdef double evaluate(self, double electron_density, double electron_temperature) except? -1e999:
```

Comment:
If this method is defined as `cdef`, It cannot be overridden in Python, only in Cython. So, if one creates a child class in Python:
```python
class ConstantLineRadiationPower(LineRadiationPower):
    """
    Constant line radiation power coefficient.
    """

    def __init__(self, value):
        self.value = value

    def evaluate(self, density, temperature):

        return self.value
```
the call:
```python
rate = ConstantLineRadiationPower(value)
val = rate(ne, te)
```
will throw a `NonImplementedError` exception because Python's `evaluate()` method in `ConstantLineRadiationPower` does not override Cython's `evaluate()` method in `_RadiatedPower`, which is not implemented.

In the `TestCase` for `TotalRadiatedPower` I create a `MockAtomicData` for which I use constant rates like the one shown above. This is just one example of why this method should be `cpdef` in the atomic rates. But there are other cases when it is useful. For example, when the atomic rates are specified analytically using the Raysect functional framework. By the way, the `evaluate()` method is specified as `cpdef` in all atomic rates except `_RadiatedPower`, `FractionalAbundance` and `TotalRadiatedPower`. So most likely in these classes this method remained `cdef` due to an oversight.

Reply:
So shouldn't it be changed to `cpdef` also in the baseclass? Allowing user to define rates in pure python is a useful option not only for tests.

Reply:
But I changed it to `cpdef` in the base class too. Otherwise it wouldn't work.

Reply:
I see, I'm sorry. I probably inspected a wrong branch when checking it. My bad. This is resolved and the PR good to go.